### DISCUSSION TITLE
KMP Compose didWriteCharacteristic() not called in BlueFalconDelegate

### DISCUSSION THREAD

**Discussion by timur-gaysin** [Reedyuk/blue-falcon#137] [Q&A]
Hi. 
I release for iOs in KMP Compose BlueFalcon

`BleDelegate: BlueFalconDelegate{
...
override fun didWriteCharacteristic(
        bluetoothPeripheral: BluetoothPeripheral,
        bluetoothCharacteristic: BluetoothCharacteristic,
        success: Boolean
    ) {
        println("didWriteCharacteristic $success")
        BLEunit.charWriteCallback(0)
    }
...
}`
`didDisconnect, didConnect, didCharacteristcValueChanged` and other successfull calls...
But when i call                         
`blueFalcon.writeCharacteristicWithoutEncoding(bluetoothPeripheral,it,data,1)` or `bluetoothPeripheral.bluetoothDevice.writeValue(data, characteristic,1)`... didWriteCharacteristic() not called. 
param characteristic i get from didDiscoverCharacteristics() and he is not null and allow for write. Whats ideas?

**Comment by Reedyuk** [comment]
Is this what you mean?
Hi,
I'm working on an iOS release using KMP Compose with BlueFalcon.
I'm implementing BleDelegate using BlueFalconDelegate:
```
override fun didWriteCharacteristic(
    bluetoothPeripheral: BluetoothPeripheral,
    bluetoothCharacteristic: BluetoothCharacteristic,
    success: Boolean
) {
    println("didWriteCharacteristic: $success")
    BLEunit.charWriteCallback(0)
}
```

The didDisconnect, didConnect, didCharacteristicValueChanged, and other callbacks are being triggered successfully.

However, when I call:

```
blueFalcon.writeCharacteristicWithoutEncoding(bluetoothPeripheral, it, data, 1)
```

or

```
bluetoothPeripheral.bluetoothDevice.writeValue(data, characteristic, 1)
```

the didWriteCharacteristic() callback is not being called.

The characteristic parameter I’m using is retrieved from didDiscoverCharacteristics()—it’s not null and is write-enabled.

Any ideas on why didWriteCharacteristic() isn’t being triggered?

**Reply by timur-gaysin** [reply]
Yes, You are right

**Comment by Reedyuk** [comment]
If the following is what you said.
Q1: Does the same behaviour happen on android?

**Reply by timur-gaysin** [reply]
No, Android module not ready yet

**Reply by Reedyuk** [reply]
you are using KMP compose, if your ios is working, android should be too.
Either-way, can you confirm it does the same on android.

**Reply by timur-gaysin** [reply]
I have working iOs code. 
On swift i call 
`peripheral.writeValue(data!, for: characteristic, type: .withResponse)` and then in
 `func peripheral(_ peripheral: CBPeripheral, didWriteValueFor characteristic: CBCharacteristic, error: Error?) {
}` i get result of write operation. 
Why i need write also Android code, if shared + iOs not work?) I think that you help me in you library...

**Reply by Reedyuk** [reply]
So its working then?
I asked which platform(ios or android) because it can help to understand where the issue could be coming from.
I'm sorry but there is only so much broken english I can support, come back with a better formatted question and I may be able to help you.

**Reply by timur-gaysin** [reply]
I wrote yesterday that problem on iOs. Sorry for my English level

**Reply by Reedyuk** [reply]
"I have working iOs code."
so its not working then?
